src/main/java/com/faforever/server/ladder/DivisionService.java

```diff
@@ -0,0 +1,144 @@
+
+import javax.annotation.PostConstruct;
+import javax.inject.Inject;
+import javax.transaction.Transactional;
+import java.text.MessageFormat;
+import java.util.Objects;
+
+@Service
+@Slf4j
+public class DivisionService {
+  private final ServerProperties properties;
+  private final DivisionRepository divisionRepository;
+  private final PlayerDivisionInfoRepository playerDivisionInfoRepository;
+
+  @Getter
+  private ImmutableList<Division> divisions;
+
+  @Inject
+  public DivisionService(ServerProperties properties, DivisionRepository divisionRepository, PlayerDivisionInfoRepository playerDivisionInfoRepository) {
+    this.properties = properties;
+    this.divisionRepository = divisionRepository;
+    this.playerDivisionInfoRepository = playerDivisionInfoRepository;
+  }
+
+  @PostConstruct
+  public void init() {
+    divisions = ImmutableList.copyOf(divisionRepository.findAllByOrderByLeagueAscThresholdAsc());
+  }
+
+  public Division getCurrentPlayerDivision(Player player) {
+    int season = properties.getLadder().getSeason();
+
+    PlayerDivisionInfo info = playerDivisionInfoRepository.findByPlayerAndSeason(player, season);
+
+    if(info == null) {
+      return null;
+    }
+
+    return divisions.stream()
+      .filter(division -> Objects.equals(division.getLeague(), info.getLeague()))
+      .filter(division -> info.getScore() <= division.getThreshold())
+      .findFirst()
+      .orElseThrow(() -> new IllegalStateException(MessageFormat.format("Could not determine division for PlayerDivisionInfo {0}", info)));
+  }
+
+  @Transactional
+  public void postResult(@NotNull Player playerOne, @NotNull Player playerTwo, @Nullable Player winner) {
+    int season = properties.getLadder().getSeason();
+
+    PlayerDivisionInfo playerOneInfo = getPlayerDivisionInfo(playerOne, season);
+    PlayerDivisionInfo playerTwoInfo = getPlayerDivisionInfo(playerTwo, season);
+
+    playerOneInfo.setGames(playerOneInfo.getGames() + 1);
+    playerTwoInfo.setGames(playerTwoInfo.getGames() + 1);
+
+    if (winner == null) {
+      log.info("Game ended in a draw - no changes in score");
+      return;
+    }
+
+    PlayerDivisionInfo winnerInfo;
+    PlayerDivisionInfo loserInfo;
+
+    if (winner.equals(playerOne)) {
+      winnerInfo = playerOneInfo;
+      loserInfo = playerTwoInfo;
+    } else if (winner.equals(playerTwo)) {
+      winnerInfo = playerTwoInfo;
+      loserInfo = playerOneInfo;
+    } else {
+      throw new IllegalArgumentException(MessageFormat.format("Winner does not match any of the players (player one: {0}, player two: {1}, winner: {2}", playerOne, playerTwo, winner));
+    }
+
+    float gain = 1.0f;
+    float loss = 0.5f;
+
+    if (winnerInfo.isInInferiorLeague(loserInfo)) {
+      gain = 1.5f;
+      loss = 1.0f;
+    } else if (winnerInfo.isInSuperiorLeague(loserInfo)) {
+      gain = 0.5f;
+      loss = 0.5f;
+    }
+
+    log.info("{} won against {} - gain: {} - loss: {}", winner, loserInfo.getPlayer(), gain, loss);
```

Comment:
Log `debug`. And we use `,` as a separator instead of `-` (applies to all log statements)